File: bridge_sdks/python/msgr_bridge_sdk/telemetry.py
"""Telemetry primitives for the Python bridge SDK."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Protocol


class TelemetryRecorder(Protocol):
    def record_delivery(self, service: str, action: str, duration: float, outcome: str) -> None:
        """Record a queue delivery metric."""


@dataclass
class NoopTelemetry:
    """Telemetry recorder that ignores all metrics."""

    def record_delivery(self, service: str, action: str, duration: float, outcome: str) -> None:
        return

File: bridge_sdks/go/msgrbridge/telemetry.go
package msgrbridge

import "time"

// TelemetryReporter records queue handling metrics.
type TelemetryReporter interface {
	RecordDelivery(service, action string, duration time.Duration, outcome string)
}

// NoopTelemetry is a TelemetryReporter that discards metrics.
type NoopTelemetry struct{}

// RecordDelivery implements TelemetryReporter.
func (NoopTelemetry) RecordDelivery(service, action string, duration time.Duration, outcome string) {}
